subsys/net/lib/nrf_cloud/src/nrf_cloud_codec.c

```diff
@@ -980,11 +981,48 @@ int nrf_cloud_obj_location_request_create(struct nrf_cloud_obj *const obj,
 		goto cleanup;
 	}
 
-	/* By default, nRF Cloud will send the location to the device */
-	if (!request_loc &&
-	    nrf_cloud_obj_num_add(&data_obj, NRF_CLOUD_LOCATION_KEY_DOREPLY, 0, false)) {
+	if (config && (obj->type == NRF_CLOUD_OBJ_TYPE_JSON)) {
```

Comment:
looking at this function, i suppose it doesn't make sense to continue if the type is not JSON, since that is the only one that is currently supported.
can you add a:
```
if (obj->type != NRF_CLOUD_OBJ_TYPE_JSON) {
		ret = -ENOTSUP;
}
```
to the beginning, and then this JSON check and the original switch statement below can be removed.